ftl/common/cache_test.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for cache.py"""
+
+import unittest
+
+import cache
+
+import mock
+
+
+class RegistryTest(unittest.TestCase):
+
+    def setUp(self):
+        pass
+
+    @mock.patch('containerregistry.client.v2_2.docker_image.FromRegistry')
```

Comment:
FWIW, a pattern I'm a big fan of is actually using `docker_build` to produce fake image data and then doing:
```python
with v2_image.FromTarball(...) as img:
   with mock.object.path(v2_image.FromRegistry, img):
      # Now FromRegistry will get "img" back.
```